gossipd/routing.h

```diff
@@ -21,6 +22,8 @@ struct node_connection {
 
 	/* Is this connection active? */
 	bool active;
+	/* Is this connection permanently inactive? */
+	bool perminactive;
```

Comment:
Permanent failure means that the channel will not be able to recover from this error, or the channel is unusable for our purposes, so we should rather remove them from our network view. That may result in us re-adding it when syncing gossip with a new peer, but we'll eventually not do the gossip dump.